File: 367. Valid Perfect Square/367. Valid Perfect Square.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        low = 1
        high = num

        while low <= high:
            mid = (low + high) // 2
            square = mid ** 2

            if square == num:
                return True
            elif square > num:
                high = mid - 1
            else:
                low = mid + 1

        return False

File: 367. Valid Perfect Square/367. Valid Perfect Square.js
/**
 * @param {number} num
 * @return {boolean}
 */
var isPerfectSquare = function (num) {
  let low = 1;
  let high = num;

  while (low <= high) {
    const mid = Math.floor((low + high) / 2);
    const square = mid * mid;

    if (square === num) {
      return true;
    } else if (square > num) {
      high = mid - 1;
    } else {
      low = mid + 1;
    }
  }

  return false;
};
